### DISCUSSION TITLE
Barchart tooltip

### DISCUSSION THREAD

**Discussion by SGEMKE** [airbnb/visx#1629] [Q&A] [answered]
Hi,

first of i want to thank you for the great library. 

I am trying to build a bar chart and currently face two issues with the tooltip. Any hints where my mistake is would be appreciated.

1. The tooltip is not updating on hover. If i enter the chart area and a new tooltip is generated it displays the correct values. However if i then hover over another bar the values inside the tooltip do not update. The correct datapoint is found since the glyph is displayed correctly.
2. The crosshair is not snapping to the datum. From my understanding it should be as simple as adding "snapTooltipToDatumX" to the tooltip. However this doesn't seem to work for me.

Here a sandbox that shows both issues:
https://codesandbox.io/s/late-hill-heophd?file=/Example.tsx

Thanks in advance!

**Comment by SGEMKE** [comment, marked as answer]
For future reference:

Both issues were caused by a wrongly set dataKey inside the AnimatedBarSeries. It took way too long but i finally figured it out.

**Reply by williaster** [reply]
hey @SGEMKE is this a bug that should be fixed/an issue that should be opened? thanks for digging in here.

**Reply by SGEMKE** [reply]
Hi @williaster ,
not a bug or issue, just wrong usage on my end. At most its a little odd usability-wise that a missing datakey will remove snapping and cause the tooltip to only update sometimes.
If you want to see the difference yourself you can replace the datakey "airfd.boundaryName" in line 103 with "airfd.mode" in the linked sandbox.
